React/state.md

```diff
@@ -1,93 +1,186 @@
 # Managing data using `state`
 
+# TODO - Update for `useState` Hook
+
 ## Learning Goals
+
 - Examine how React allows components to manage data
 - Use `state` within a component
 - Examine how `props` and `state` affect component rendering
 
 ## `state`
-State is managed **within a given component**. State can be accessed using the `this.state` object and it can be modified using the `this.setState()` function.
 
-#### Overview
-- Can be assigned to any object key you want
-- `setState` will always take an object as the parameter
-- `setState` will merge any params passed in with the existing state data, it won't overwrite it
-- We often set the initial state in the constructor function
+State is managed **within a given component**. State can be accessed using the `useState` hook and it can be modified using the a function that `useState` returns.
+
+**Wait what's a hook?**  A hook is a special function react provides to "hook" into the lifecyle of a React component and access specific functionality.  The `useState` hook function allows us to have a variable, like an instance variable for a Ruby class for our component.  It also provides a function to change the value of that variable, or it's state.  When this state variable is changed, the component is re-rendered.  React has a number of hooks, and you can even create your own, but for now, we will focus on `useState`.
```

Comment:
```suggestion
**Wait what's a hook?**  A hook is a special function react provides to "hook" into the lifecyle of a React component and access specific functionality.  The `useState` hook function allows us to have a variable with scope beyond the execution of a function. In Ruby, an instance variable served this same purpose.  It also provides a function to change the value of that variable (i.e. it's state).  When this state variable is changed, the component is re-rendered.  React has a number of hooks, and you can even create your own, but for now, we will focus on `useState`.
```

Reply:
This is my attempt to expand on the comparison to instance variables a little. But feel free to ignore if you don't feel it's more clear.